WordPress/Classes/ViewRelated/Post/Scheduling/SchedulingCalendarViewController.swift

```diff
@@ -0,0 +1,199 @@
+        if let date = coordinator?.date {
+            datePicker.date = date
+        }
+        self.datePicker = datePicker
+        return datePicker
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        chosenValueRow.titleLabel?.text = NSLocalizedString("Choose a time", comment: "Label for Publish time picker")
+        chosenValueRow.detailLabel?.text = datePicker.date.longStringWithTime()
+        let doneButton = UIBarButtonItem(title: NSLocalizedString("Done", comment: "Label for Done button"), style: .done, target: self, action: #selector(done))
+        navigationItem.setRightBarButton(doneButton, animated: false)
+    }
+
+    @IBAction func timePickerChanged(_ sender: Any) {
+        chosenValueRow.detailLabel?.text = datePicker.date.longStringWithTime()
+        coordinator?.setDate(datePicker.date)
+    }
+
+    @IBAction func done() {
+        coordinator?.updated(coordinator?.date)
+        navigationController?.dismiss(animated: true, completion: nil)
+    }
+}
+
+/// A base class used by the date picker classes
+/// Could be genericized further
+class DatePickerSheet: UIViewController {
+    var coordinator: DateCoordinator? = nil
+
+    weak var chosenValueRow: ChosenValueRow!
+    weak var pickerView: UIView!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        WPStyleGuide.configureColors(view: view, tableView: nil)
+
+        let chosenValueRow = configureValueRow()
+        let pickerView = configureView()
+        pickerView.translatesAutoresizingMaskIntoConstraints = false
+
+        let pickerWrapperView = UIView()
+        pickerWrapperView.addSubview(pickerView)
+
+        let sideConstraints: [NSLayoutConstraint] = [
+            pickerView.leftAnchor.constraint(equalTo: pickerWrapperView.leftAnchor),
+            pickerView.rightAnchor.constraint(equalTo: pickerWrapperView.rightAnchor)
+        ]
+
+        // Allow these to break on larger screen sizes and just center the content
+        sideConstraints.forEach() { constraint in
+            constraint.priority = .defaultHigh
+        }
+
+        NSLayoutConstraint.activate([
+            pickerView.centerXAnchor.constraint(equalTo: pickerWrapperView.safeCenterXAnchor),
+            pickerView.topAnchor.constraint(equalTo: pickerWrapperView.topAnchor),
+            pickerView.bottomAnchor.constraint(equalTo: pickerWrapperView.bottomAnchor)
+        ])
+
+        NSLayoutConstraint.activate(sideConstraints)
+
+        let stackView = UIStackView(arrangedSubviews: [
+            chosenValueRow,
+            pickerWrapperView
+        ])
+        stackView.axis = .vertical
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+
+        self.chosenValueRow = chosenValueRow
+        self.pickerView = pickerView
+
+        view.addSubview(stackView)
+        view.pinSubviewToSafeArea(stackView)
+    }
+
+    // Does nothing, should be overriden by the subclass
+    func configureView() -> UIView {
```

Comment:
The [Swift API Guidelines](https://swift.org/documentation/api-design-guidelines/) recommend to:

> Begin names of factory methods with “make”, e.g. x.makeIterator().

So this would be best as `makeView` – however, I think it could also be more specific: `makePickerView`, to clarify that the method isn't used to create the entire view for the view controller.